deeplake/core/transform/transform.py

```diff
@@ -212,42 +221,87 @@ def my_fn(sample_in: Any, samples_out, my_arg0, my_arg1=0):
 
         initial_autoflush = target_ds.storage.autoflush
         target_ds.storage.autoflush = False
-        progress_end_args = {"compute_id": compute_id, "progress": 100, "end": True}
 
         if not check_lengths:
             skip_ok = True
 
-        try:
-            self.run(
-                data_in,
-                target_ds,
-                compute_provider,
-                num_workers,
-                scheduler,
-                progressbar,
-                overwrite,
-                skip_ok,
-                read_only_ok and overwrite,
-                **kwargs,
-            )
-            target_ds._send_compute_progress(**progress_end_args, status="success")
-        except Exception as e:
-            target_ds._send_compute_progress(**progress_end_args, status="failed")
-            raise TransformError(e).with_traceback(sys.exc_info()[2])
-        finally:
-            compute_provider.close()
+        if checkpoint_interval > 0:
+            if num_workers > 0 and checkpoint_interval % num_workers != 0:
+                raise ValueError(
+                    "checkpoint_interval should be a multiple of num_workers if num_workers > 0"
+                )
             if overwrite:
-                original_data_in.storage.clear_cache_without_flush()
-                load_meta(original_data_in)
-                if pad_data_in and not initial_padding_state:
-                    original_data_in._disable_padding()
-                if not kwargs.get("disable_rechunk"):
-                    rechunk_if_necessary(original_data_in)
-            else:
-                load_meta(target_ds)
-                target_ds.storage.autoflush = initial_autoflush
-                if not kwargs.get("disable_rechunk"):
-                    rechunk_if_necessary(target_ds)
+                raise ValueError(
+                    "checkpoint_interval > 0 and ds_out is None. Cannot checkpoint during inplace transform."
+                )
+            datas_in = [
+                data_in[i : i + checkpoint_interval]
+                for i in range(0, len(data_in), checkpoint_interval)
+            ]
+
+        else:
+            datas_in = [data_in]
+
+        samples_processed = 0
+        desc = get_pbar_description(self.functions)
+        pbar = get_progressbar(len(data_in), desc)
+        pqueue = compute_provider.create_queue()
+        for i, data_in in enumerate(datas_in):
+            progress = int((i + 1) / len(datas_in) * 100)
```

Comment:
@AbhinavTuli FYI this is mathematically slightly incorrect because each element in the list does not have the same number of samples. Not sure if it's work fixing, but if possible+simple, we should print the sample_level progress.

Reply:
fixed

Reply:
although this progress was for the events we send to backend which used to be binary 0 and 100% earlier, now we send more events, the tqdm is still sample level